File: archives/20190519python/0080. Remove Duplicates from Sorted Array II.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if not nums:
            return 0
        if len(nums)==1:
            return 1
        p1=0
        p2=1
        cnt=1
        while p2<len(nums):
            if nums[p2]!=nums[p1]:
                p1+=1
                nums[p1]=nums[p2]
                cnt=1
                p2+=1
            else:
                if cnt==1:
                    cnt+=1
                    p1+=1
                    nums[p1]=nums[p2]
                    p2+=1
                else:
                    p2+=1
        return p1+1

File: archives/C#/0080. Remove Duplicates from Sorted Array II.cs
public class Solution {
    public int RemoveDuplicates(int[] nums) {
        if(nums.Length!=0){
            int left=0;
            int target=nums[0];
            int cnt=0;
            for(int i=0;i<nums.Length;i++){
                if(nums[i]==target && cnt!=2){
                    nums[left]=target;
                    cnt++;
                    left++;
                }
                else if(nums[i]==target && cnt==2){
                    continue;
                }
                else{
                    target=nums[i];
                    nums[left]=target;
                    left++;
                    cnt=1;
                }
            }
            return left;
        }
        return 0;
    }
}
